src/pages/experiments/[experimentId]/data-processing/index.jsx

```diff
@@ -101,18 +102,47 @@ const DataProcessingPage = () => {
       <Col span='8'>
         <Select
           value={stepId}
-          onChange={(id) => setStepId(id)}
+          onChange={(id) => {
+            setStepId(id);
+          }}
           style={{ width: 360, fontWeight: 'bold' }}
           placeholder='Jump to a step...'
         >
           {
             steps.map(
-              ({ name, id }, i) => (
+              ({ name, key }, i) => (
                 <Option
                   value={i}
-                  disabled={!completedSteps.has(id)}
+                  key={key}
+                  disabled={!completedSteps.has(key)}
                 >
-                  {name}
+
+                  {completedSteps.has(key) && (
+                    <Text
+                      type='success'
+                    >
+                      <CheckOutlined />
+                      <span style={{ marginLeft: '0.25rem' }}>{name}</span>
+                    </Text>
+                  )}
+
+                  {!completedSteps.has(key) && stepId === i && (
+                    <Text
+                      type='default'
+                    >
+                      <CaretRightOutlined />
+                      <span style={{ marginLeft: '0.25rem' }}>{name}</span>
+                    </Text>
+                  )}
+
+                  {!completedSteps.has(key) && stepId !== i && (
+                    <Text
+                      disabled
+                    >
+                      <EllipsisOutlined />
+                      <span style={{ marginLeft: '0.25rem' }}>{name}</span>
+                    </Text>
+                  )}
```

Comment:
Some UX changes that make it easier to navigate the dropdown menu regarding which steps are done, outstanding, currently filling.